nuxeo-services/nuxeo-invite/src/main/java/org/nuxeo/ecm/user/invite/DefaultInvitationUserFactory.java

```diff
@@ -79,4 +84,36 @@ public NuxeoPrincipal doCreateUser(CoreSession session, DocumentModel registrati
         }
         return user;
     }
+
+    /**
+     * Filter group by computing the intersection of the group in the registration doc and the groups of the user that
+     * created the request. Administrators accept all groups.
+     *
+     * @param registrationDoc
+     * @param configuration
+     *
+     * @since 10.2
+     */
+    private List<String> filterGroups(DocumentModel registrationDoc, UserRegistrationConfiguration configuration) {
+        List<String> wantedGroup = (List<String>) registrationDoc.getPropertyValue(
+                configuration.getUserInfoGroupsField());
+
+        String originatingUser = registrationDoc.getProperty(UserInvitationComponent.PARAM_ORIGINATING_USER)
+                                                .getValue(String.class);
```

Comment:
This works but as mostly nobody uses it we'll probably deprecate it at some point. Try to use `getPropertyValue` instead, even if it requires a cast (which the generic version just hides).